hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/handler/CloseRegionHandler.java

```diff
@@ -101,6 +105,19 @@ public void process() throws IOException {
         return;
       }
 
+      rsServices.getBlockCache().ifPresent(blockCache -> {
+        if (blockCache instanceof CombinedBlockCache) {
+          BlockCache l2 = ((CombinedBlockCache)blockCache).getSecondLevelCache();
+          if (l2 instanceof BucketCache) {
+            if (region.getReadOnlyConfiguration().get(PREFETCH_PERSISTENCE_PATH_KEY) != null) {
+              LOG.info("Closing region {} during a graceful stop, and prefetch persistence is on, "
+                + "so setting evict on close to false. ", region.getRegionInfo().getEncodedName());
+              region.getStores().forEach(s -> s.getCacheConfig().setEvictOnClose(false));
+            }
+          }
+        }
+      });
+
```

Comment:
I know I've done something similar in UnassignRegionHandler for HBASE-27474, but I wonder now if we should move these logic about evict decision to a related methods in HRegion.